_rules/SC3-1-2-lang-valid.md

```diff
@@ -0,0 +1,119 @@
+---
+name: Valid body lang attribute
+
+description:
+- This rule checks that `lang` or `xml:lang` attributes on elements within the `body` of a web page conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).
+
+success_criterion:
+- 3.1.2
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Bryn Anderson
+- Jey Nandakumar
+---
+
+## Test Procedure
+
+### Applicability
+
+Any DOM element, within the `body` of a webpage with a [non-empty](#non-empty) `lang` or `xml:lang` attribute.
+
+**Note**: To retrieve all DOM elements with a `lang` or `xml:lang` attribute, other than the root element use -
+`descendent-or-self::body[@lang]` -or-  `descendent-or-self::body[@xml:lang]`.
```

Comment:
Is there a reason for why an XPath selector has been included in this rule? Seems like an implementation detail that the rule should't concern itself with.